src/Dockerfile

```diff
@@ -0,0 +1,23 @@
+ARG     PYVER=3
+FROM    python:${PYVER}
+
+# Enable unbuffered STDOUT logging
+ENV     PYTHONUNBUFFERED=1
```

Comment:
Why do we need this out of interest? And is it something we should be doing even outside of Docker?

Reply:
I/O buffering is done at multiple levels for better performance such as operating systems and run time environments. Usually, the behavior of programs depends on where the STDOUT is pointed to. If we write to a file, data is usually written in blocks of fixed size and the OS/runtime waits for at least one block of data to be ready before it writes out, unless it is flushed explicitly. When the output is connected to a TTY, it yields data on each newline. This makes debugging easier, but when running in production, it is advised to log in files instead of a terminal for better performance.

When running processes in Docker, OS-level buffering is handled properly, depending on whether or not the output is connected to a TTY, but due to an increased level of indirection, Python's runtime does not get to know it, so it continues to buffer output internally. As result, when trying to print a debug message in a process running in a container attached to a virtual TTY, we might not see anything being logged on terminal for a while until there is enough data to yield. That's why, I have added this environment variable to tell Python not to enable I/O buffering. I will not do the same in a production environment. #LessonsLearnedTheHardWayOverTime ;-)

Reply:
OK so a Docker-specific thing. Cool. Resolving.